js/core/src/Blocks.svelte

```diff
@@ -227,13 +227,45 @@
 		return get_data(component_id);
 	}
 
+	function waitForInputReady(dependency: Dependency): Promise<void> {
+		return new Promise((resolve) => {
+			const checkInputsReady = (): boolean => {
+				return dependency.inputs.every((inputId) => {
+					const component = components.find(
+						(comp) => comp.id === Number(inputId)
+					);
+					return component?.props["input_ready"] === true;
```

Comment:
@pngwn Is there a better way to check for a components props? This doesn't seem to work as expected.